### PR TITLE
Update expression language5.0 to10.1.8

### LINKED ISSUE #24752: Update Expression Language 5.0 to latest 10.1.8 version
**Describe the bug**  
A new version of the Tomcat Expression Language API / Implementation is available. Update from the current 10.1.5 version to the latest 10.1.8 version.

**Steps to Reproduce**  
N/A

**Expected behavior**  
N/A

**Diagnostic information:**  
N/A

**Additional context**  
[API](https://repo1.maven.org/maven2/org/apache/tomcat/tomcat-el-api/10.1.8/)
[IMPL](https://repo1.maven.org/maven2/org/apache/tomcat/tomcat-jasper-el/10.1.8/)

### PR DESCRIPTION
**Pull request by pnicolucci**
fixes #24752

### CODE REVIEW

**Comment by pnicolucci** [comment]
#libby #build

**Comment by pnicolucci** [comment]
#libby #build

**Comment by pnicolucci** [comment]
#libby #build

**Comment by pnicolucci** [comment]
#libby #build

**Comment by pnicolucci** [comment]
#libby #build

**Comment by pnicolucci** [comment]
#libby #build

**Review by volosied — APPROVED** [review]
looks good to me

File: dev/io.openliberty.org.apache.jasper.expressionLanguage.5.0_fat/test-applications/EL50Coercion.war/src/io/openliberty/el50/fat/coercion/servlets/lambda/EL50LambdaExpressionCoercionServlet.java [hunk 1]
```diff
@@ -68,30 +60,34 @@ public void testLambdaCoercion() throws Exception {
...
-        }
-        assertEquals("The expression did not throw expected ELException. Got result: "+result, true, exceptionCaught);
+        String result;
+        String expectedResult = "failed";
+
+        Object obj = elp.eval("lambdaBean.testIntLambdaCoercion(param -> param.equals('should be int not string'))");
+        assertNotNull(obj);
+        result = obj.toString();
+
+        assertEquals("The expression did not evaluate to: " + expectedResult + " but was: " + result, expectedResult, result);
```
Comment (volosied) [inline dev/io.openliberty.org.apache.jasper.expressionLanguage.5.0_fat/test-applications/EL50Coercion.war/src/io/openliberty/el50/fat/coercion/servlets/lambda/EL50LambdaExpressionCoercionServlet.java:81]: Linking to explanation of change: https://lists.apache.org/thread/cm0pycoyqq4pb1j816vdpdo4nds57dwj 9e (expand Mark's comment) 

NPE occurs in this test, but was masked.

**Pull request merged**